.github/workflows/release_java.yml

```
@@ -0,0 +1,216 @@
+name: Release Java
+run-name: Release ${{github.event.inputs.release-version || github.ref_name}}
+concurrency:
+  group: ${{github.workflow}}-${{github.ref}}
+  cancel-in-progress: true
+on:
+  push:
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+-java" # ex. v1.0.0-java
+      - "v0.0.1-rc[0-9]+-java" # used for testing only
```

I think I found the issue. The `\+` fails inside of the `"` due to yaml. Putting it into single quotes `'` works

```suggestion
      - 'v[0-9]+.[0-9]+.[0-9]+\+java[0-9]+' # ex. v1.0.0+java1
      - 'v0.0.1-rc[0-9]+\+java[0-9]+' # used for testing only

```

---

Great discovery! I'm still surprised why that is the case.